include/sys/zio.h

```diff
@@ -140,7 +140,7 @@ enum zio_encrypt {
 #define	ZIO_COMPRESS_LEGACY_ON_VALUE	ZIO_COMPRESS_LZJB
 #define	ZIO_COMPRESS_LZ4_ON_VALUE	ZIO_COMPRESS_LZ4
 
-#define	ZIO_COMPRESS_DEFAULT		ZIO_COMPRESS_OFF
+#define	ZIO_COMPRESS_DEFAULT		ZIO_COMPRESS_ON
```

Comment:
As part of making this change we're going to also need to update any ZFS Test Suite test case which assume that compression is disabled by default.

Reply:
For future reference:
During my work on the compression side of the test suite, "ON" was removed from the list of tested algorithms. change of this shouldn't have any influence on those tests anymore :)

Reply:
it is mistake to do `ZIO_COMPRESS_DEFAULT=ZIO_COMPRESS_ON` by default because root pool = rpool = boot pool can be/should be uncompressed - for example: on SPARC with openboot.
we are on DilOS is using rpool on intel and SPARC are uncompressed, but setup compression=on to dataset for BE(boot environment)

Reply:
@ikozhukhov Very good argument.
So you would suggest not doing this at all?

Just leave it to OS vendors/creators (downstream) to decide when to use compression?

Reply:
yes, it is very easy to do:
```
zpool create -O compression=on testpool vdev1
```
no issues, but try to do it `off` every time in installer is not good way - it's my opinion.

Reply:
@behlendorf If agreed uppon I think its best to close this PR completely because:
1. If someone wants to do something like this, it needs WAY more work than one line.
2. This PR is just 1 obvious line and is useless to keep open.

Reply:
I think there is a reasonable argument to be made that enabling compression by default would be generally a good thing.  And that depending on the compression algorithm selected this wouldn't necessarily be an issue for most (maybe all?) of the ZFS-aware bootloaders.

That said, I do think there would need to be some additional investigation before we could move forward with this.  So let's close this out for now, we can always revisit this.